### PR TITLE
Untangle view letters visibility logic

### PR DESCRIPTION
**Pull request by bshyong**
## Description
If countries or states are unavailable, and users click "edit address", they end up in a state where they can no longer view letters. The button is disabled due to being in Edit Mode, and the user is stuck and cannot exit that mode. 

The button logic has been modified so that it is disabled when in address editing mode, but not if the user is in editing mode but ends up in a terminal error state.

## Testing done
Local testing

## Screenshots
Screen Shot 2019-06-19 at 2 16 46 PM

## Acceptance criteria
- [x] View letters should be available to users who should have access, even if Edit Address is unavailable

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by narin — DISMISSED** [review]
LGTM, but unit tests are now failing.

File: src/applications/letters/containers/AddressSection.jsx [hunk 1]
```diff
@@ -372,14 +381,35 @@ export class AddressSection extends React.Component {
...
-          {addressContent}
+        <div>
+          <Element name="addressScrollElement" />
+          <div aria-live="polite" aria-relevant="additions">
+            {/* Warning message goes here while editing with no address on record */}
+            {emptyAddress && this.props.isEditingAddress && noAddressBanner}
+            {addressContent}
+          </div>
         </div>
+        {viewLettersButton}
```
Comment (hinzed1127) [inline src/applications/letters/containers/AddressSection.jsx]: I think you could add the `vads-u-display--block` [utility class](https://design.va.gov/utilities/display) here and avoid the need for nesting another div

File: src/applications/letters/containers/AddressSection.jsx [hunk 2]
```diff
@@ -361,6 +367,9 @@ export class AddressSection extends React.Component {
       addressContent = (
         <div className="step-content">{addressUpdateUnavailable}</div>
       );
+      // override needed for view letters button so that users can
+      // still view letters if they end up in this terminal state
+      forceEnableViewLettersButton = true;
```
Comment (hinzed1127) [inline src/applications/letters/containers/AddressSection.jsx]: would this ever get set to false? just wondering if this is the clearest say to designate the weird UX

Comment (hinzed1127) [reply src/applications/letters/containers/AddressSection.jsx]: nvm I realize that the initialization is falsy, maybe just explicitly set it to false above (line 360)

Comment (bshyong) [reply src/applications/letters/containers/AddressSection.jsx]: yep I'll do that for clarity

**Review by narin — APPROVED** [review]

**Pull request merged**